tidy3d/components/tcad/boundary/heat.py

```diff
@@ -40,6 +53,137 @@ class HeatFluxBC(HeatChargeBC):
     )
 
 
+class NaturalConvectionVerticalSpec(HeatChargeBC):
+    """
+    Specification for natural convection from a vertical plate.
+
+    This class calculates the heat transfer coefficient (h) based on fluid
+    properties and an expected temperature difference, then provides these
+    values as 'base' and 'exponent' for a generalized heat flux equation
+    q = base * (T_surf - T_fluid)^exponent.
+    """
+
+    # --- Input Parameters ---
+    fluid_k: pd.NonNegativeFloat = pd.Field(
+        title="Fluid Thermal Conductivity",
+        description="Thermal conductivity (k) of the fluid.",
+        units=THERMAL_CONDUCTIVITY,
+    )
+    fluid_mu: pd.NonNegativeFloat = pd.Field(
+        title="Fluid Dynamic Viscosity",
+        description="Dynamic viscosity (μ) of the fluid.",
+        units=DYNAMIC_VISCOSITY,
+    )
+    fluid_Cp: pd.NonNegativeFloat = pd.Field(
+        title="Fluid Specific Heat",
+        description="Specific heat of the fluid at constant pressure.",
+        units=SPECIFIC_HEAT,
+    )
+    fluid_rho: pd.NonNegativeFloat = pd.Field(
+        title="Fluid Density",
+        description="Density (ρ) of the fluid.",
+        units=DENSITY,
+    )
+    fluid_beta: pd.NonNegativeFloat = pd.Field(
+        title="Fluid Thermal Expansivity",
+        description="Thermal expansion coefficient (β) of the fluid.",
+        units=THERMAL_EXPANSIVITY,
+    )
+    plate_L: pd.NonNegativeFloat = pd.Field(
+        title="Plate Characteristic Length",
+        description="Characteristic length (L), defined as the height of the vertical plate.",
+        units=MICROMETER,
```

Comment:
should all fluid properties be pulled automatically from medium properties that touches that boundary? Sounds like we should expand heat_spec with these additional properties. For other parameters I would use more explicit names: `plate_L` -> `plate_length`